### PR TITLE
[4.2.01]: changelog update 

### PR DESCRIPTION
**Pull request by masterleinad**
Includes: 
- #6626 
- #6631 
- #6746
- #6627 
- #6638 
- #6655
- https://github.com/kokkos/kokkos/pull/6723
- #6749
- #6742

### CODE REVIEW

**Comment by ndellingwood** [comment]
#6723 includes cleanup of whitespace of the CHANGELOG.md, merge of that PR back into these updates should propagate the cleanup

**Comment by ndellingwood** [comment]
Rebased on top of https://github.com/kokkos/kokkos/pull/6723

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,24 @@
...
+#### HIP:
+- Fix Graph compilation error [\#6624](https://github.com/kokkos/kokkos/pull/6624)
+- Add missing gfx940 [\#6671](https://github.com/kokkos/kokkos/pull/6671)
+
+#### SYCL:
+- Fix testing with `bfloat16` support [\#6608](https://github.com/kokkos/kokkos/pull/6608)
+
+### Bug Fixes
+- Remove deprecation warning for `AllocationMechanism` for `gcc` <11.0 [\#6653](https://github.com/kokkos/kokkos/pull/6653)
+- Avoid finalizing profiling in execution space instances [\#6635](https://github.com/kokkos/kokkos/pull/6635)
```
Comment (dalg24) [inline CHANGELOG.md:20]:
```suggestion
- Fix bug early tools finalize with non-default host execution instances [\#6635](https://github.com/kokkos/kokkos/pull/6635)
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,24 @@
...
+## [4.2.01](https://github.com/kokkos/kokkos/tree/4.2.01) (2023-12-07)
+[Full Changelog](https://github.com/kokkos/kokkos/compare/4.2.00...4.2.01)
+
+### Backend and Architecture Enhancements:
+
+#### CUDA:
+- Add warp sync for `parallel_reduce` to avoid race condition [\#6630](https://github.com/kokkos/kokkos/pull/6630)
+
+#### HIP:
+- Fix Graph compilation error [\#6624](https://github.com/kokkos/kokkos/pull/6624)
```
Comment (dalg24) [inline CHANGELOG.md:12]:
```suggestion
- Fix Graph "multiple definition of" linking error (missing `inline` specifier) [\#6624](https://github.com/kokkos/kokkos/pull/6624)
```

**Review by dalg24 — COMMENTED** [review]
The MSVC+CUDA build fix from #6660 is missing

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,5 +1,24 @@
...
+- Add missing gfx940 [\#6671](https://github.com/kokkos/kokkos/pull/6671)
+
+#### SYCL:
+- Fix testing with `bfloat16` support [\#6608](https://github.com/kokkos/kokkos/pull/6608)
```
[the remark this answers is not in the document]
Comment (ndellingwood) [reply CHANGELOG.md]: Dropped in 25995897d7a0dcaec770916de1c8ad23222fed6c

**Comment by ndellingwood** [comment]
> The MSVC+CUDA build fix from #6660 is missing

Added in 25995897d7a0dcaec770916de1c8ad23222fed6c

File: CMakeLists.txt [hunk 4]
```diff
@@ -151,7 +151,7 @@ ENDIF()
 
 set(Kokkos_VERSION_MAJOR 4)
 set(Kokkos_VERSION_MINOR 2)
-set(Kokkos_VERSION_PATCH 0)
+set(Kokkos_VERSION_PATCH 01)
```
Comment (dalg24) [inline CMakeLists.txt:154]:
```suggestion
set(Kokkos_VERSION_PATCH 1)
```

File: Makefile.kokkos [hunk 5]
```diff
@@ -2,7 +2,7 @@
 
 KOKKOS_VERSION_MAJOR = 4
 KOKKOS_VERSION_MINOR = 2
-KOKKOS_VERSION_PATCH = 0
+KOKKOS_VERSION_PATCH = 01
```
Comment (dalg24) [inline Makefile.kokkos:5]:
```suggestion
KOKKOS_VERSION_PATCH = 1
```

File: CHANGELOG.md [hunk 6]
```diff
@@ -1,5 +1,26 @@
...
+- Add support for gfx940 (AMD Instinct MI300 GPU) [\#6671](https://github.com/kokkos/kokkos/pull/6671)
+
+### Build System
+- CMake: Don't let Kokkos set `CMAKE_CXX_FLAGS` for Trilinos builds [\#6742](https://github.com/kokkos/kokkos/pull/6742)
+
+### Bug Fixes
+- Remove deprecation warning for `AllocationMechanism` for `gcc` <11.0 [\#6653](https://github.com/kokkos/kokkos/pull/6653)
+- Fix bug early tools finalize with non-default host execution instances [\#6635](https://github.com/kokkos/kokkos/pull/6635)
+- Fix various issues for MSVC CUDA builds [\#6659](https://github.com/kokkos/kokkos/pull/6659)
+- Remove extra `;` in Kokkos_SIMD_Scalar.hpp: [\#6510](https://github.com/kokkos/kokkos/pull/6510)
```
Comment (dalg24) [inline CHANGELOG.md:22]:
```suggestion
- Fix "extra `;`" warning with `-pedantic` flag in `<Kokkos_SIMD_Scalar.hpp>` [\#6510](https://github.com/kokkos/kokkos/pull/6510)
```

File: CHANGELOG.md [hunk 7]
```diff
@@ -1,5 +1,26 @@
...
+
+#### HIP:
+- Fix Graph "multiple definition of" linking error (missing `inline` specifier) [\#6624](https://github.com/kokkos/kokkos/pull/6624)
+- Add support for gfx940 (AMD Instinct MI300 GPU) [\#6671](https://github.com/kokkos/kokkos/pull/6671)
+
+### Build System
+- CMake: Don't let Kokkos set `CMAKE_CXX_FLAGS` for Trilinos builds [\#6742](https://github.com/kokkos/kokkos/pull/6742)
+
+### Bug Fixes
+- Remove deprecation warning for `AllocationMechanism` for `gcc` <11.0 [\#6653](https://github.com/kokkos/kokkos/pull/6653)
```
Comment (dalg24) [inline CHANGELOG.md:19]:
```suggestion
- Remove deprecation warning for `AllocationMechanism` for GCC <11.0 [\#6653](https://github.com/kokkos/kokkos/pull/6653)
```

**Review by dalg24 — APPROVED** [review]

**Review by ndellingwood — APPROVED** [review]

**Comment by ndellingwood** [comment]
@dalg24 @masterleinad @crtrott if all updates for the patch release are complete, we just need this PR merged then I can start the snapshot to Trilinos and put up PRs to master

**Pull request merged**